src/_blog/cryptotls.md

```diff
@@ -0,0 +1,52 @@
+---
+tags:
+- quic
+- go
+- standard library
+- crypto/tls
+title: Adding a QUIC API for Go's standard library TLS package
+description:
+date: 2023-08-15
+permalink: "/2023-08-15-quic-crypto-tls/"
+translationKey: ''
+header_image: /quic-crypto-tls.png
+author: Marten Seemann
+---
+
+# Adding a QUIC API for Go's standard library TLS package
+
+You might have [heard us talk about this before](https://www.youtube.com/watch?v=aDHymXQJ4bs): QUIC has become the most important transport in libp2p. For example, QUIC accounts for 80-90% of the connections made to PL-run bootstrappers participating in the public IPFS DHT (see our [public go-libp2p dashboard](https://protocollabs.grafana.net/public-dashboards/2bd3f1bee9964d40b6786fbe3eafd9fc?orgId=1)). [quic-go](https://github.com/quic-go/quic-go) is a QUIC implementation written in pure Go that is not only used in go-libp2p, but also in [Caddy](https://caddyserver.com/), a full-featured webserver (for its HTTP/3 support), [Adguard](https://github.com/AdguardTeam/AdGuardHome), for providing DNS over QUIC, and in the synchronization tool [syncthing](https://github.com/syncthing/syncthing/), and [many other projects](https://github.com/quic-go/quic-go#projects-using-quic-go).
```

Comment:
```suggestion
You might have [heard us talk about this before](https://www.youtube.com/watch?v=aDHymXQJ4bs): QUIC has become the most important transport in go-libp2p. For example, QUIC accounts for 80-90% of the connections made to PL-run bootstrappers participating in the public IPFS DHT (see our [public go-libp2p dashboard](https://protocollabs.grafana.net/public-dashboards/2bd3f1bee9964d40b6786fbe3eafd9fc?orgId=1)). [quic-go](https://github.com/quic-go/quic-go) is a QUIC implementation written in pure Go that is not only used in go-libp2p, but also in [Caddy](https://caddyserver.com/), a full-featured webserver (for its HTTP/3 support), [Adguard](https://github.com/AdguardTeam/AdGuardHome), for providing DNS over QUIC, and in the synchronization tool [syncthing](https://github.com/syncthing/syncthing/), and [many other projects](https://github.com/quic-go/quic-go#projects-using-quic-go).
```

Is it really the most important in libp2p or go-libp2p; there's a difference. Of the 12 libp2p implementations, how many support quic apart from Go and Rust? I don't think this claim is the right one to make

Reply:
12 implementations? That's a very optimistic to count them. But if you want to go down that route, there are even some QUIC-only libp2p implementations among them, for example zig-libp2p.

I think this statement is broader than just the respective implementations. QUIC is the transport that we're optimizing for when designing new protocols / new features. It's nice if a new feature _somehow_ works on TCP, but QUIC is the thing we care about.

Reply:
zig is not widely deployed though. I still strongly favor you stating "QUIC is becoming the most important transport in libp2p" - I think this is more accurate given that it is in the process of being enabled in the Beacon chain.

Reply:
"is becoming" sgtm